### PR TITLE
installer response file names may have the same name

### LINKED ISSUE #159: Image creations fails when the response files have the same name from different location.
--installerResponseFile /home/brkarthi/source/docker-images/OracleFMWInfrastructure/dockerfiles/12.2.1.4.0/install.file,/home/brkarthi/source/docker-images/OracleWebCenterSites/dockerfiles/12.2.1.4.0/install.file

When using the above option, image creation fails in copying of response file

### PR DESCRIPTION
**Pull request by ddsharpe**
Fixes #159 
When installer response file names originate from different directories, it is possible that they will have the same simple filename.  When copying those response files to the Docker context folder, rename the target file as needed to accommodate any number of response files that have the same "original" simple name.

### CODE REVIEW

File: imagetool/src/main/java/com/oracle/weblogic/imagetool/installer/ProvidedResponseFile.java [hunk 1]
```diff
@@ -16,25 +16,40 @@
...
 
     /**
      * Use one of the default response files provided in the tool.
      * @param source the resource path to the response file in this tool
      */
     public ProvidedResponseFile(Path source) {
         responseFileSource = source;
+        if (responseFileSource != null && Files.isRegularFile(responseFileSource)) {
+            filename = responseFileSource.getFileName().toString();
+        }
```
Comment (jshum2479) [inline imagetool/src/main/java/com/oracle/weblogic/imagetool/installer/ProvidedResponseFile.java:29]: What if the source is null or not a file? name() will fail

File: imagetool/src/main/java/com/oracle/weblogic/imagetool/installer/ProvidedResponseFile.java [hunk 2]
```diff
@@ -16,25 +16,40 @@
...
     @Override
     public String name() {
-        return responseFileSource.getFileName().toString();
+        return filename;
     }
 
     @Override
     public void copyFile(String buildContextDir) throws IOException {
-        if (responseFileSource != null && Files.isRegularFile(responseFileSource)) {
+        if (filename != null) {
```
Comment (jshum2479) [inline imagetool/src/main/java/com/oracle/weblogic/imagetool/installer/ProvidedResponseFile.java:39]: may be calling name() instead?

**Comment by markxnelson** [comment]
I don't understand the purpose of this - don't you need to know the name of the response file later in order to use it? Or at least to know which one is which?

**Comment by ddsharpe** [comment]
> I don't understand the purpose of this - don't you need to know the name of the response file later in order to use it? Or at least to know which one is which?

Yes, it does.  The response files are tied to the installer. Normally, the default response file is sufficient.  But if the user wants to override the default response file that Image Tool provides, they must specify a silent installer response file for each installer they use.  For example, if they ask to install SOA, they need two response files, one for the SOA installer, and one for the FMW installer.  If those installer response files come from different directories, it is possible that they have the same file name.  So when they get copied to the Docker build context directory with the same filename, there was an issue.

**Pull request merged**